pkg/generators/echoapi/generator.go

```diff
@@ -17,40 +17,65 @@ const (
 
 // Generator configures the generators for EchoAPI
 type Generator struct {
-	generators.BaseOptions
-	Spec saasv1alpha1.EchoAPISpec
+	generators.BaseOptionsV2
+	Spec    saasv1alpha1.EchoAPISpec
+	Traffic bool
 }
 
 // NewGenerator returns a new Options struct
-func NewGenerator(instance, namespace string, spec saasv1alpha1.EchoAPISpec) Generator {
+func NewGenerator(instance, namespace string, spec saasv1alpha1.EchoAPISpec) (Generator, error) {
 	return Generator{
-		BaseOptions: generators.BaseOptions{
-			Component:    component,
-			InstanceName: instance,
-			Namespace:    namespace,
-			Labels: map[string]string{
-				"app":     component,
-				"part-of": "3scale-saas",
+			BaseOptionsV2: generators.BaseOptionsV2{
+				Component:    component,
+				InstanceName: instance,
+				Namespace:    namespace,
+				Labels: map[string]string{
+					"app":     component,
+					"part-of": "3scale-saas",
+				},
 			},
+			Spec:    spec,
+			Traffic: true,
 		},
-		Spec: spec,
+		nil
+}
+
+// Validate that Generator implements workloads.TrafficManager interface
+var _ workloads.TrafficManager = &Generator{}
+
+func (gen *Generator) Services() []basereconciler_resources.ServiceTemplate {
+	return []basereconciler_resources.ServiceTemplate{
+		{Template: gen.service(), IsEnabled: true},
+	}
+}
+func (gen *Generator) SendTraffic() bool { return gen.Traffic }
```

Comment:
You could have just returned always `true` as we are not implementing canary here. The Traffic field in the Generator struct is not really needed.

Reply:
Sure no problem, let's leave it like this.